### ISSUE TITLE
Unable to see metrics in UI when using external prometheus

### ISSUE THREAD

**Issue by m1nish1208** [linkerd/linkerd2#8309]
### What is the issue?

Environment Details: 
2 node cluster hosted on Azure VPC
Linkerd Control Plane version: 2.1.11
Linkerd Client Version: 2.1.11

I followed instructions specified here --> https://linkerd.io/2.11/tasks/install-helm/  & https://linkerd.io/2.11/tasks/external-prometheus/

When I login to Dashboard I don't see any metrics for applications that are meshed. The error I get in the logs is as follows:
[
![2](https://user-images.githubusercontent.com/9932114/164513315-0ee27a8e-a4d6-4762-b55f-0e8c8b26aabd.png)
![1](https://user-images.githubusercontent.com/9932114/164513343-263159f6-749b-4b9b-a2c7-bfcd7e1dfc3c.png)
](url)
 
If I use linked viz CLI I can see metrics but I believe thats because of tap extension.

### How can it be reproduced?

Use external prometheus instance & disable built-in.

### Logs, error output, etc

Attached screenshots.

### output of `linkerd check -o short`

```
Linkerd core checks
===================
 
linkerd-version
---------------
‼ cli is up-to-date
    is running version 2.11.1 but the latest stable version is 2.11.2
    see https://linkerd.io/2.11/checks/#l5d-version-cli for hints
 
control-plane-version
---------------------
‼ control plane is up-to-date
    is running version 2.11.1 but the latest stable version is 2.11.2
    see https://linkerd.io/2.11/checks/#l5d-version-control for hints
 
linkerd-control-plane-proxy
---------------------------
‼ control plane proxies are up-to-date
    some proxies are not running the current version:
        * linkerd-identity-d8d6d7cc7-fss2r (stable-2.11.1)
        * linkerd-destination-75f4785568-qdnzs (stable-2.11.1)
        * linkerd-proxy-injector-65975c5c67-b2prr (stable-2.11.1)
    see https://linkerd.io/2.11/checks/#l5d-cp-proxy-version for hints
 
Status check results are √
 
linkerd-vizensions                              - Running viz extension check
-----------                            \ Running viz extension check
‼ viz extension proxies are up-to-date
    some proxies are not running the current version:
        * grafana-685c7998d8-q6rgs (stable-2.11.1)
        * web-f4dc8bd5f-zntmc (stable-2.11.1)
        * metrics-api-5bb9564dcd-cpm7h (stable-2.11.1)
        * tap-6d979599c9-hxnhw (stable-2.11.1)
        * tap-injector-cdffbd494-pqcjj (stable-2.11.1)
        * prometheus-75ff4cc4-ctblc (stable-2.11.1)
    see https://linkerd.io/2.11/checks/#l5d-viz-proxy-cp-version for hints
 
Status check results are √
```

### Environment

- Kubernetes Version: microk8s 1.22 
- Cluster Environment: Azure 
- Host OS: Ubuntu 20.04
- Linkerd version: 
```
Client version: stable-2.11.1
Server version: stable-2.11.1
```

**Comment by olix0r** [comment]
> If I use linked viz CLI I can see metrics but I believe thats because of tap extension.

What CLI commands are you using? `linkerd viz stat` should use prometheus data.

Have you confirmed that your prometheus instance is scraping proxies successfully? 

It's difficult to make sense of screenshots; but I think this is showing that requests are timing out to prometheus.

**Comment by m1nish1208** [comment]
Hi.. 

I've been able to fix this.. I think I misread  https://linkerd.io/2.11/tasks/external-prometheus/. 

I'm using prometheus provided by kube-prometheus-operator, while investigating additionalScrapeConfigs.yaml (the https://github.com/prometheus-community/helm-charts/blob/main/charts/kube-prometheus-stack/templates/prometheus/additionalScrapeConfigs.yaml) I found that it doesn't have config for it to scrape Linkerd metrics. Once I added those missing configs, I can see the stats. 

Thanks, I think we can close this.
